stores/directory/local.py

```diff
@@ -23,42 +26,81 @@ def __init__(self, local_dir: Union[str, Path]):
         assert local_dir_path.is_dir(), f"Given path {local_dir_path} is not a directory."
 
         # Store that location against the class
-
         self.local_path = local_dir_path
 
-
     def add_file(self, file: Path):
         # Abstract methods can remain without implementation for now.
         pass
 
     def has_lone_file_matching(self, pattern: str) -> bool:
-        #  To be implemented in local directory store 1 #5 
-        ...
+        # Grab a list of files matching the regex pattern to determine how many exist.
+
+        matching_files = self._files_that_match_pattern(pattern)
+
+        if len(matching_files) == 1:  # 1 file matched
+            return True
+        elif len(matching_files) == 0:  # 0 file matched
+            return False         
+        else:  # 2+ files matched
+            raise FileNotFoundError(f"More than 1 file found that matches the regex pattern '{pattern}' in directory {self.local_path}. Matching: {matching_files}")
+
+    def save_lone_file_matching(self, pattern: str, destination: Optional[Union[Path, str]] = None):
+        # Assert 1 file matches
+        if not self.has_lone_file_matching(pattern):
+            raise FileNotFoundError(f"No matching files found for pattern {pattern} in directory {self.local_path}")
+
+        file_to_save = self._files_that_match_pattern(pattern)[0]
+        file_name = Path(file_to_save).name
+
+        if destination is not None:
+            if isinstance(destination, str):
+                destination = Path(destination)
+            
+            save_path = Path(destination / file_name)
+        else:
+            save_path = Path(file_name)
+
+        if save_path.exists():
+            raise ValueError(f"Given file already exists in directory {save_path}")
+
+        with open(file_to_save) as f:
+            file_data = f.read()
+
+        with open(save_path, "w") as f:
+            f.write(file_data)
 
-    def save_lone_file_matching(self, pattern: str, save_as: str) -> Path:
-        #  To be implemented in local directory store 1 #5 
-        ...
 
     def get_lone_matching_json_as_dict(self, pattern: str) -> dict:
-        #  To be implemented in local directory store 1 #5 
-        ...
+        # Assert 1 file matches
+        if self.has_lone_file_matching(pattern): 
+            file_path = Path(self.local_path / self._files_that_match_pattern(pattern)[0])
+
+            # use json.load to put contents of file into variable and return dict.
+            with open(file_path) as f:
+                json_dict = json.load(f)
+            return json_dict
 
     def save_lone_file_matching_regex(self, regex_str) -> Path:
```

Comment:
can you remove this one and the method from the base if its in there, it looks like I duplicated save_lone_file_matching